src/pip/_internal/commands/install.py

```diff
@@ -237,9 +237,6 @@ def __init__(self, *args, **kw):
     @with_cleanup
     def run(self, options, args):
         # type: (Values, List[Any]) -> int
```

Comment:
I think it's worth keeping it as a contract to prevent regressions:
```suggestion
        # type: (Values, List[Any]) -> int
        assert not options.use_user_site or options.target_dir is None, "Can not combine '--user' and '--target'"
```

Reply:
@McSinyx ?